### PR TITLE
Multitiles poddoors + doors code cleanup

### PR DESCRIPTION
**Pull request by Menshin**
## About The Pull Request

* Made poddoors use inherited procs instead of rewriting them when possible
* made the door base class correctly handles 4-tiles doors regarding opacity and boundaries box
* multitile opacity is now handled by an `opacifier` object instead of meddling with turfs opacity directly
* extended `UNACIDABLE` and `INDESTRUCTIBLE` flags to all multitiles poddoors (instead of only the 4 tiles horizontal one)
* minor code cleaning (`locked`, `welded` and `not_weldable` are now attributes of the door base class, lots of defines replacing)

### CODE REVIEW

File: code/game/machinery/doors/poddoor.dm [hunk 1]
```diff
@@ -1,381 +1,73 @@
-
 /obj/machinery/door/poddoor
 	name = "\improper Podlock"
 	desc = "That looks like it doesn't open easily."
 	icon = 'icons/obj/doors/rapid_pdoor.dmi'
 	icon_state = "pdoor1"
-	id = 1.0
-	dir = 1
+	id = ""
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]: null?

File: code/game/machinery/doors/poddoor.dm [hunk 2]
```diff
@@ -1,381 +1,73 @@
...
+	dir = SOUTH
```
Comment (spookydonut) [inline code/game/machinery/doors/poddoor.dm]: dont all objects have a default dir?

**Comment by Zostroll** [comment]
I spent so long on these doors and you went and did it all better than me :(

File: code/game/machinery/doors/poddoor.dm [hunk 3]
```diff
@@ -1,381 +1,71 @@
...
 		spawn(0)
-			operating = 1
-			flick("pdoorc0", src)
-			icon_state = "pdoor0"
-			SetOpacity(0)
+			operating = TRUE
 			sleep(15)
-			density = 0
-			operating = 0
+			open()
```
Comment (Rohesie) [inline code/game/machinery/doors/poddoor.dm]: Any change of a `CALLBACK` here, to remove those `spawn` and `sleep` procs?

Comment (Menshin) [reply code/game/machinery/doors/poddoor.dm]: Sure.

**Pull request merged**